src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -255,7 +255,7 @@ public static Assembly LoadFile(string path)
                 throw new ArgumentException(SR.Format(SR.Argument_AbsolutePathRequired, path), nameof(path));
             }
 
-            string normalizedPath = Path.GetFullPath(path);
+            string normalizedPath = Path.GetFullyQualifiedPath(path);
```

Comment:
This is behavior change. 

If somebody calls `Assembly.LoadFile("c:\\a\0a")` today, they will get `ArgumentException: Illegal characters in path.` exception. After this change, we will proceed to load file called `c:\a`. Is it intentional? Other places touched in this PR have same problem.